src/App.jsx

```diff
@@ -14,45 +20,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1[sortField] - good2[sortField];
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState();
+  const visibleGoods
+    = getPreparedGoods(goodsFromServer, { sortField, reverseField });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          className={cn('button is-info', { 'is-light':
+            sortField !== SORT_FIELD_NAME })}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+          type="button"
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          className={cn('button is-success', { 'is-light':
+            sortField !== SORT_FIELD_LENGTH })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+        >
+          Sort by length
+        </button>
+
+        <button
+          className={cn('button is-warning', { 'is-light':
+            !reverseField })}
+          onClick={() => setReverseField(!reverseField)}
+          type="button"
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverseField)
+        && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReverseField(undefined);
+            }}
+          >
+            Reset
+          </button>
+        )}
```

Comment:
```suggestion
        {(sortField || reverseField) && (
          <button
            type="button"
            className="button is-danger is-light"
            onClick={() => {
              setSortField('');
              setReverseField(undefined);
            }}
          >
            Reset
          </button>
        )}
```
Also, we can create a variable flag for `sortField || reverseField`